docs/src/boundary_conditions_doc.md

```diff
@@ -0,0 +1,165 @@
+# Boundary Conditions (BCs)
+
+This section is an attempt to bridge mathematical considerations of BCs and their implementations in climate models.
```

Comment:
I think this doc is great and very helpful, but may perhaps belong with the RQA work in ClimateMachine. What do you think?

Reply:
RQA said they will add it once they've revamped their BCs, so perhaps we can leave it here until then?